### PR TITLE
fix: update integration test to work as presubmit check

### PR DESCRIPTION
**Pull request by nolanmar511**
Kokoro indicates the PR being checked by PR number.
This change modified the integration test so PR number can be used to check out source code during the integration test.

### CODE REVIEW

File: testing/integration_test.go [hunk 1]
```diff
@@ -103,19 +105,31 @@ type nodeGCETestCase struct {
 
 func (tc *nodeGCETestCase) initializeStartUpScript(template *template.Template) error {
 	var buf bytes.Buffer
+
+	var checkoutBranchCmd string
+	if *pr != 0 {
+		checkoutBranchCmd = fmt.Sprintf(
+			`git fetch origin pull/%d/head:pull_branch
```
Comment (aalexand) [inline testing/integration_test.go]: This became a strange mix of using templates and formatting strings manually to my taste. I'd suggest passing both pr and branch to the template and use template conditions to format the commands the needed way. See "if" pipeline in https://golang.org/pkg/text/template/#hdr-Actions.

File: testing/presubmit.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
```
Comment (aalexand) [inline testing/presubmit.sh]: You could probably share some pieces of this file with continuous.sh. Or use the same script and condition slightly depending on whether KOKORO_GITHUB_PULL_REQUEST_NUMBER is set?

File: testing/integration_test.go (hunk 1 above)
Comment (nolanmar511) [reply testing/integration_test.go]: Done.

File: testing/presubmit.sh (hunk 2 above)
Comment (nolanmar511) [reply testing/presubmit.sh]: Done.

File: testing/integration_test.go [hunk 3]
```diff
@@ -65,9 +66,15 @@ npm -v
 node -v
 
 # Install agent
-git clone https://github.com/GoogleCloudPlatform/cloud-profiler-nodejs.git
+git clone {{.Repo}}
 cd cloud-profiler-nodejs
-git pull {{.Repo}} {{.Branch}}
+{{if .PR}}
```
Comment (aalexand) [inline testing/integration_test.go]: I think you could do simpler:
```
git fetch origin pull/{{if .PR}}{{.PR}}{{else}}{{.Branch}}{{end}}/head:pull_branch
git checkout pull_branch
```

Comment (nolanmar511) [reply testing/integration_test.go]: done.

**Comment by nolanmar511** [comment]
PTAL

**Pull request merged**